torchgen/native_function_generation.py

```diff
@@ -359,14 +428,17 @@ def add_generated_native_functions(
             # There are only two functions that don't fit this criteria today though,
             # and they both look like they should be fixed to be out= variants,
             # so if feels safer to ban this schema all-together
-            gets_out_variant = not has_out and (
-                base_fn.func.kind() == SchemaKind.inplace
-                or any(r.type.is_tensor_like() for r in base_fn.func.returns)
+            base_fn_valid = base_fn.func.kind() == SchemaKind.inplace or any(
+                r.type.is_tensor_like() for r in base_fn.func.returns
             )
-            if not has_out and not gets_out_variant:
+
+            gets_out_variant = not has_out and base_fn_valid
+            if not has_out and not base_fn_valid:
                 if (
                     str(base_fn.func.name)
                     not in MUTABLE_OPS_THAT_CANNOT_GET_AN_OUT_VARIANT
+                    and str(base_fn.func.name)
+                    not in FUNCTIONAL_OPS_THAT_CANNOT_GET_AN_OUT_VARIANT
                 ):
                     raise AssertionError(
                         f"""Found a mutable operator that we could not generate an out= variant for: {str(base_fn.func)}.
```

Comment:
Nit: "Found a mutable operator" -> "Found an operator", now that the op we're generating out= variants of can be functional too.

It's probably worth updating this error message a bit too: if someone adds a new functional op that returns a non-tensor output, the error message should probably tell them to directly add their op name to the `FUNCTIONAL_OPS_THAT_CANNOT_GET_AN_OUT_VARIANT` list.